src/envoy/http/authn/test_utils.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "common/protobuf/protobuf.h"
+#include "gmock/gmock.h"
+#include "src/envoy/http/authn/context.pb.h"
+
+namespace Envoy {
+namespace Http {
+namespace IstioAuthN {
+
+std::unique_ptr<Payload> CreateX509Payload(const std::string& user) {
```

Comment:
Do we need to use so many unique_ptr?

try to use as many stack memory as possible?

like

Create(.... Payload* );

Reply:
Done